back-end/helpers/intent_analyzer.spec.js

```diff
@@ -0,0 +1,59 @@
+import TestIntentGenerator from "./test-helpers/testIntentGenerator.js";
+import IntentAnalyzer from "./intent_analyzer.js"
+
+let analyzer;
+let generator;
+
+const getPercentages = ((children, total_children) => {
+  const percentageMap = {}
+    children.forEach((numTimesCalled, intent) => {
+      percentageMap[intent] = (numTimesCalled / total_children).toFixed(2);
+    })
+    return percentageMap;
+})
+
+describe("intentAnalyzer", () => {
+  beforeEach(() => {
+    analyzer = new IntentAnalyzer();
+    generator = new TestIntentGenerator();
+  });
+
+  it("Should return an empty array if given no model", async() => {
+    expect(analyzer.analyzeIntents([])).toStrictEqual([]);
+  });
+
+  it("Should properly format one intent", async() => {
+    const fakeMap = new Map();
+    fakeMap.set("Intent 1-DOT-", 1);
+    const input = generator.generateModel("Question 1-DOT-", fakeMap, 1, 1);
+    input.getPercentages.mockImplementation(() => {return getPercentages(input.children, input.total_children)})
+    const output = generator.generateAnalyzedIntent("Question 1.", {"Intent 1.": 1.00});
+
+    expect(analyzer.analyzeIntents([input])).toStrictEqual([output]);
+  });
+
+  it("Should properly format three intents", async() => {
+    const fakeMap1 = new Map();
+    fakeMap1.set("Intent 1-DOT-", 7);
+    const fakeMap2 = new Map();
+    fakeMap2.set("Intent 2-DOT-", 2);
+    const fakeMap3 = new Map();
+    fakeMap3.set("Intent 3-DOT-", 1);
```

Comment:
Can we make a test for a map with multiple children to do a rough check of our percent calc